source_code/main_mcu/src/LOGIC/logic_fido2.c

```diff
@@ -366,49 +374,33 @@ static uint32_t logic_fido2_make_auth_data_existing_cred(fido2_make_auth_data_re
     /* Sanity checks */
     /*****************/
     _Static_assert(sizeof(response->tag) == sizeof(credential_id), "tag size is too large");
-    _Static_assert(sizeof(response->user_ID) >= sizeof(user_handle), "user handle is too large");
+    _Static_assert(sizeof(response->user_handle) >= sizeof(user_handle), "user handle is too large");
     _Static_assert(sizeof(response->rpID_hash) == sizeof(auth_data_header.rpID_hash), "rpid hash is too large");
-    _Static_assert(sizeof(response->pub_key_x) == sizeof(pub_key.x), "pub key x is too large");
-    _Static_assert(sizeof(response->pub_key_y) == sizeof(pub_key.y), "pub key y is too large");
     _Static_assert(sizeof(response->aaguid) == sizeof(fido2_minible_aaguid), "aaguid is too large");
     
+
+    /* Sanitize user_handle_len to prevent overflow */
+    if (user_handle_len > MEMBER_SIZE(fido2_get_assertion_rsp_message_t, user_handle))
+    {
+        user_handle_len = MEMBER_SIZE(fido2_get_assertion_rsp_message_t, user_handle);
+    }
+
     /*************************/
     /* Create answer to host */
     /*************************/
     memcpy(response->tag, credential_id, sizeof(credential_id));
-    memcpy(response->user_ID, user_handle, sizeof(user_handle));
+    memcpy(response->user_handle, user_handle, user_handle_len);
+    response->user_handle_len = user_handle_len;
     memcpy(response->rpID_hash, auth_data_header.rpID_hash, sizeof(auth_data_header.rpID_hash));
     response->count_BE = auth_data_header.sign_count;
-    memcpy(response->pub_key_x, pub_key.x, sizeof(pub_key.x));
-    memcpy(response->pub_key_y, pub_key.y, sizeof(pub_key.y));
```

Comment:
how come this is not needed anymore? :)

Reply:
guessing it's due to the different messages used :)

Reply:
correct. The make credential need this. The get assertion does not.